examples/plot_boxplot.py

```diff
@@ -12,6 +11,6 @@
         "iris.csv",
     )
 
-conn = create_engine("duckdb://")
+conn = Connection.from_connect_str("duckdb://").session
```

Comment:
check out the master branch, I modified the examples to use `sqlalchemy.create_engine`, that's b better than using our internal Connection object.

Reply:
I figured out that if I use `create_engine` way, it will cause the build to fail, that's why initially I used this `Connection.from_connect_str`, [error build](https://readthedocs.org/projects/jupysql/builds/20038606/)

Reply:
The root cause is in the `_transpile_query()` method will lookup the current Connection dialect, because we are using `create_engine` here so it still sees the SQLite is the current dialect (which should be duckdb)

1. The reason SQLite still remain is probably the last execution code remain on it's states
2.  The `create_engine` here wouldn't reset the current connection, only `Connection.from_connect_str` can

What do you think? I agree it's weird to have `Connection.from_connect_str` in the example python file since it's not the good practice, but the problem here is it wouldn't pass the Read the Docs build

Probably we can skip the Read the Docs build on those example files?

Reply:
Ok, I understand what's the problem now. This is a design problem.

I opened https://github.com/ploomber/jupysql/issues/386

Also, I looked at the code and looks like this PR is making the boxplot compatible with other vendors since we're now transpiling the query. If so, let's change the name of this PR since it's misleading. Even if we don't see big performance improvements, I think we should merge this. we can tackle the performance tweaks later.

@idomic: we found a design problem that I consider we should tackle before merging this one.